test/system/src/test/java/io/pravega/test/system/AbstractReadWriteTest.java

```diff
@@ -552,6 +573,89 @@ private EventWriterConfig buildWriterConfig() {
         return validEvents;
     }
 
+    void createScope(String scopeName, StreamManager streamManager) {
+        Boolean createScopeStatus = streamManager.createScope(scopeName);
+        if (createScopeStatus) {
+            log.info("Successfully created Scope {}", scopeName);
+        }
+    }
+
+    void createKVT(String scopeName, String kvtName, KeyValueTableConfiguration config, URI controllerURIDirect) {
+        log.info("Creating KVT name:{} in scope {}", kvtName, scopeName);
+        Boolean createKvtStatus = KeyValueTableManager.create(controllerURIDirect).createKeyValueTable(scopeName, kvtName, config);
+        if (createKvtStatus) {
+            log.info("Successfully created KVT {}", kvtName);
+        }
+    }
+
+    void startKVPCreate(String scope, String kvtName, KeyValueTableFactory keyValueTableFactory, URI controllerURIDirect) {
+        testInsertUpdateGetKVP(scope, kvtName, keyValueTableFactory, controllerURIDirect);
+    }
+
+    private void testInsertUpdateGetKVP(String scope, String kvtName, KeyValueTableFactory keyValueTableFactory, URI controllerURIDirect) {
+        log.info("Start KVP operation for KVT {} of scope {}", kvtName, scope);
+        // Creating 5 threads to update KVP table entry
+        for (Integer i = 1; i < 6; i++) {
+            String value = convertStringToBinary("Hello World");
+            Integer keyId = i;
+            CompletableFuture.runAsync(() -> {
```

Comment:
Are you keeping any reference to these `CompletableFuture`s? If you see how we do things for regular or transactional writers, we keep a reference to these futures to verify when these tasks complete and complete successfully (or not).

Reply:
Adding the CompletableFuture as a reference, and test will be fail if task not complete